Mathlib/Algebra/MvPolynomial/Degrees.lean

```diff
@@ -282,6 +281,21 @@ theorem degreeOf_mul_X_eq (j : σ) (f : MvPolynomial σ R) :
   convert Multiset.count_le_of_le j (degrees_X' (R := R) j)
   rw [Multiset.count_singleton_self]
 
+theorem degreeOf_mul_X_eq_degreeOf_add_one_iff (j : σ) (f : MvPolynomial σ R) :
+    degreeOf j (f * X j) = degreeOf j f + 1 ↔ f ≠ 0:= by
+  refine ⟨fun h => by by_contra ha; simp [ha] at h, fun h => ?_⟩
+  apply Nat.le_antisymm (degreeOf_mul_X_eq j f)
+  have : (f.support.sup fun m ↦ m j) + 1 = (f.support.sup fun m ↦ (m j + 1)) := by
```

Comment:
I'm surprised this auxiliary proof is as long as it is. I'd expect there to be some theorem `g (s.sup f) = s.sup (g ⚬ f)` for monotone functions `g`. Maybe you can add that to the PR?

Reply:
@vihdzp 
added `comp_sup_eq_sup_comp_of_nonempty`. One really not only needs monotone, but also nonempty, since otherwise it is just not true for `Bot`.

From my point of view this is now ready for review.